src/Common/src/System/Collections/Generic/LargeArrayBuilder.cs

```diff
@@ -216,25 +231,35 @@ public CopyPosition CopyTo(CopyPosition position, T[] array, int arrayIndex, int
             int row = position.Row;
             int column = position.Column;
 
-            for (; count > 0; row++, column = 0)
+            T[] buffer = GetBuffer(row);
+            int copied = CopyToCore(buffer, column);
+
+            if (count == 0)
             {
-                T[] buffer = GetBuffer(index: row);
+                return new CopyPosition(row, column + copied).Normalize(buffer.Length);
+            }
 
-                // During this iteration, copy until we satisfy `count` or reach the
-                // end of the current buffer.
-                int copyCount = Math.Min(buffer.Length, count);
+            do
+            {
+                buffer = GetBuffer(++row);
+                copied = CopyToCore(buffer, 0);
+            } while (count > 0);
 
-                if (copyCount > 0)
-                {
-                    Array.Copy(buffer, column, array, arrayIndex, copyCount);
+            return new CopyPosition(row, copied).Normalize(buffer.Length);
 
-                    arrayIndex += copyCount;
-                    count -= copyCount;
-                    column += copyCount;
-                }
-            }
+            int CopyToCore(T[] sourceBuffer, int sourceIndex)
```

Comment:
Moving this to `private int CopyToCore(T[], int, T[] array, int arrayIndex, int count)` in the class lets the build command pass on my dev box. Seems like a product bug that compiling this crashes on OSX.

Reply:
Are we using different compiler versions in release/2.0 vs master?

Reply:
Looking at `Tools/tool-runtime/project.csproj` `Microsoft.Net.Compilers.netcore`:
 * release/2.0.0: `2.0.0-rc`
 * master: `2.0.0-rc4`

Reply:
@dagood, do you know why we're still using old(er) preview bits?  At the very least I would have thought we'd be using a shipped compiler, or if not, it'd be because we're using something cutting edge.

@OmarTawfik, in the name of moving this forward, can you avoid using the local function?

@agocke, you implemented local functions, right?  Does this crash sound familiar at all?

Reply:
> @dagood, do you know why we're still using old(er) preview bits?

There's an upgrade going on in https://github.com/dotnet/buildtools/pull/1661, but I don't have any broader context on why certain versions are being used.

Reply:
@stephentoub I'll move to a class function for now.